src/init/starship.bash

```diff
@@ -24,7 +24,7 @@ starship_preexec() {
 # Will be run before the prompt is drawn
 starship_precmd() {
     # Save the status, because commands in this pipeline will change $?
-    STATUS=$?
+    local STATUS=$? PIPE="${PIPESTATUS[@]}";
```

Comment:
It looks when using `bash-preexec` this will have to be `BP_PIPESTATUS`: https://github.com/rcaloras/bash-preexec/blob/master/bash-preexec.sh#L41-L45